### DISCUSSION TITLE
Why do some methods take out parameters instead of returning values?

### DISCUSSION THREAD

**Discussion by markonius** [bepu/bepuphysics2#175] [General]
Some methods, such as `Box.ComputeInertia` and `Bodies.GetDescription` take an `out` parameter instead of returning a value. Why is that?

**Comment by RossNordby** [comment]
Historically, large struct copy avoidance, occasionally other usability considerations depending on the function.

ComputeInertia is one API in particular that tends to be *very* annoying as a result. Modern runtimes have significantly improved the costs involved with the direct return, too. It's something I've been intending to poke at.
